mllib/src/main/scala/org/apache/spark/mllib/linalg/Vectors.scala

```diff
@@ -556,6 +579,28 @@ class SparseVector(
       i += 1
     }
   }
+
+  override def hashCode(): Int = {
```

Comment:
OK, sounds fine.  Curious: Why do DenseVector and SparseVector need to implement the same hash code?  I was thinking DenseVectors could stay as they are (the first 16 elements), while SparseVectors could look at the first 16 non-zeros to reduce collisions.  But I'm not familiar with this code.